util/log/clog.go

```diff
@@ -705,6 +724,53 @@ func (l *loggingT) output(s severity, buf *buffer, file string, line int, alsoTo
 	}
 }
 
+// processForStderr colors log lines depending on severity.
+func (l *loggingT) processForStderr(s severity, data []byte) []byte {
+	switch l.color {
+	case "auto":
+		if !l.checkForColorTerm() {
+			return data
+		}
+	case "always":
+		// Colorize no matter what.
+	default:
+		return data
+	}
+	idx := bytes.Index(data, []byte("]"))
+	if idx == -1 {
+		return data
+	}
+	var prefix []byte
+	switch s {
+	case warningLog:
+		prefix = []byte("\033[0;31;49m") // red
+	case errorLog, fatalLog:
+		prefix = []byte("\033[1;31;49m") // bold & red
+	default:
+		return data
+	}
+	prefix = append(prefix, data[:idx+1]...)
+	prefix = append(prefix, []byte("\033[0m")...)
+	return append(prefix, data[idx+1:]...)
+}
+
+// checkForColorTerm attempts to verify that stderr is a character
+// device and if so, that the terminal supports color output.
+func (l *loggingT) checkForColorTerm() bool {
+	if l.hasColorTerm == nil {
+		var color bool
+		fi, _ := os.Stderr.Stat() // get the FileInfo struct describing the standard input.
+		if (fi.Mode() & os.ModeCharDevice) != 0 {
+			term := os.Getenv("TERM")
+			if strings.Contains(term, "xterm") && strings.Contains(term, "color") {
```

Comment:
I don't want to go too far down this rabbit hole, but TERM=ansi also seems to be somewhat standard and that's what my emacs shells use.